Docs: Distinguish examples in rules under Stylistic Issues part 2

scriptdaemon opened the pull request:
The second batch of (5) rules under Stylistic Issues.

The next batch will be bigger; I just didn't have as much time today as I had hoped. Depending on how this PR goes, I might just add the next 5-10 directly to this instead of a separate batch.

@pedrottimark There were a lot of differences in these rules from your latest updates, so could you give this a thorough look?

One concern before this gets merged: For options that don't have an explicit name, where should we put the code examples (apologies if this is answered in one of your issues / PRs)?

I found some more inconsistencies between some rules here and other rules, e.g. for the `id-match` rule, the `properties` object property option takes either `true` or `false`, but in other rules it takes `always` or `never`. This is outside of the scope of documentation updates, however, and would also be a breaking change, but perhaps something to keep in mind for the future.

keithamus commented on docs/rules/id-blacklist.md:
```diff
@@ -1,41 +1,29 @@
-- function calls (so you can still use functions you do not have control over)
-- object properties (so you can still use objects you do not have control over)
-
+This rule disallows specified identifiers in assignments and `function` definitions.
```
This sentence doesn't feel like as complete of a description as the original.

pedrottimark replied:
@keithamus What do you think of something like the following to keep the contrasting bulleted lists but decrease the length of sentences?

This rule disallows specified identifiers in:
* variable declarations
* function declarations
* assignment statements [however to make it clear the lhs including object properties]

This rule does not apply to identifiers which you might not have control over:
* function calls
* object properties [however to make it clear other than assignment mentioned above]

keithamus replied:
I think that is great 😄

pedrottimark commented:
@scriptdaemon Another branch works well for next batch. Otherwise both you and the reviewers might feel like the goalposts keep moving. My first PR for rule docs had 40 comments for 11 files 🙃

Thank you for lending a hand.

A vision to aim for is rule docs that are
1. correct
2. clear
3. complete for primary ESLint audience
4. patterns are robust and sustainable for future rules and options when busy developers will need to imitate similar precedents to the best of their ability

pedrottimark commented:
To build toward your question step by step:
* For rules like no-control-regex without options, examples are under Rule Details.
* For rules like no-console with limited exception options, my current thought is baseline examples are under Rule Details and additional correct code is under Options.
* For rules like this batch, examples are under Options but without h3 headings if there is no string literal or property key. I had started to study these rules and remember they have some rare or even unique option schema.

pedrottimark commented:
What do you think about making the main h1 heading as similar as possible to the corresponding item on the rules index page, so there is a more obvious visual connection after you click a link?

Even if you arrive at a rule doc from search results or formatted error report, it seems helpful to lead with the rule id. See https://github.com/eslint/eslint/issues/5774#issuecomment-208396092 for a picture. Your comments on that proposal are welcome.

Because some of the descriptions have code style, my instinct is away from traditional heading capitalization (or even sentence capitalization) to limit an appearance of visual complexity.

pedrottimark commented:
Another was to carefully invest visual complexity is remove double quote marks or back ticks from options in h3 headings:
* to counter-balance the inline code to specify options in example sentences
* in case docs ever gets autogenerated table of contents of links to their h2 and h3 headings

scriptdaemon commented:
@keithamus Re-added more of the description. I wasn't sure how simplified we wanted to go with that section, but it appears previous pull requests do allow for description for built-in exceptions, etc.

@pedrottimark Updated PR with your suggestions.

Changing the `h1` header to be of the form `rule: description` will have to come later (otherwise it breaks the tests), but I do agree with keeping the title lowercase.

The quotes I left in the `h3` headers were a mistake, and I have removed them. I agree they are not needed.

Lastly, I also added the term `additionally` for rule exceptions. Let me know if there are other cases where it should also apply.

pedrottimark commented:
@scriptdaemon Thanks. You are on the leading edge of progress again.

Very nice attention to detail improving grammar, removing unnecessary empty lines in example code, and also adding missing semicolons.

While proofreading, I imagined `"id-blacklist": ["error", ["foo", "bar", "baz"]]` to lint example code, haha.

Oops, we’ll update descriptions in h1 headings now, but let the other change wait until later. My bad.
